src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -0,0 +1,156 @@
+        bool set{ false };
+    };
+}
+
+// Use this macro to quickly implement both getters and the setter for an
+// inheritable and observable setting property. This is similar to the GETSET_PROPERTY macro, except...
+// - Has(): checks if the user explicitly set a value for this setting
+// - Getter(): return the resolved value
+// - Setter(): set the value directly
+// - Clear(): clear the user set value
+// - the setting is saved as an optional, where nullopt means
+//   that we must inherit the value from our parent
+#define GETSET_SETTING(type, name, ...)                                     \
+public:                                                                     \
+    /* Returns true if the user explicitly set the value, false otherwise*/ \
+    bool Has##name() const                                                  \
+    {                                                                       \
+        return _##name.has_value();                                         \
+    };                                                                      \
+                                                                            \
+    /* Returns the resolved value for this setting */                       \
+    /* fallback: user set value --> inherited value --> system set value */ \
+    type name() const                                                       \
+    {                                                                       \
+        const auto val{ _get##name##Impl() };                               \
+        return val ? *val : type{ __VA_ARGS__ };                            \
+    };                                                                      \
+                                                                            \
+    /* Overwrite the user set value */                                      \
+    /* Dispatch event if value changed */                                   \
+    void name(const type& value)                                            \
+    {                                                                       \
+        if (_##name != value)                                               \
+        {                                                                   \
+            _##name = value;                                                \
+        }                                                                   \
+    };                                                                      \
+                                                                            \
+    /* Clear the user set value */                                          \
+    /* Dispatch event if value changed */                                   \
+    void Clear##name()                                                      \
+    {                                                                       \
+        if (Has##name())                                                    \

```

Comment:
no need to check; cheap to assign nullopt over nullopt

Reply:
Ah. So the if-statement in "Clear" and the setter are for the _PropertyChanged dispatcher.

Reply:
Removed since we may not need these to be observable anyways. And making them observable shouldn't be too difficult to add later.